collector/gke.go

```diff
@@ -11,57 +11,67 @@ import (
 
 	"github.com/DazWilkin/gcp-exporter/gcp"
 	"github.com/prometheus/client_golang/prometheus"
-
 	"google.golang.org/api/cloudresourcemanager/v1"
 	"google.golang.org/api/container/v1"
 	"google.golang.org/api/googleapi"
 )
 
+// GKECollector collects GKE cluster metrics
 type GKECollector struct {
-	account               *gcp.Account
-	enableExtendedMetrics bool
-
-	Info          *prometheus.Desc
-	NodePoolsInfo *prometheus.Desc
-	Nodes         *prometheus.Desc
-	Up            *prometheus.Desc
+	account                         *gcp.Account
+	enableExtendedMetrics           bool
+	extraLabelsClusterInfoExtended  ExtraLabel
+	extraLabelsNodePoolsInfoExtended ExtraLabel
+
+	info          *prometheus.Desc
+	nodePoolsInfo *prometheus.Desc
+	nodes         *prometheus.Desc
+	up            *prometheus.Desc
 }
 
-func NewGKECollector(account *gcp.Account, enableExtendedMetrics bool) *GKECollector {
+// NewGKECollector initializes the GKE collector
+func NewGKECollector(account *gcp.Account, enableExtendedMetrics bool, extraLabelsClusterInfo string, extraLabelsNodePoolsInfo string) *GKECollector {
 	fqName := name("gke")
 	labelKeys := []string{"project", "name", "location", "version"}
 
+	// Process the extra labels once
+	extraLabelsClusterInfoMap := ProcessExtraLabels(extraLabelsClusterInfo)
```

Comment:
Gate these on their presence:

```golang
if len(extraLabelsClusterInfo)>0 {...}
if len(extraLabelsNodePoolsInfo)>0 {...}
```
The functions capture this but it's more evident what's being done with the checks.

Reply:
and I think this should include the extension of the label sets:

```golang
infoLabelKeys := labelKeys
if len(extraLabelsFoo)>0 {
    infoLabels = append(infoLabels, GetLabelNamesFromExtraLabels(...))
}
...

info: prometheus.NewDesc(
   ...
   infoLabelKeys
)
```